fixed Windows backend (for now)

tm9k1 opened the pull request:
## open for reviews and discussions!

tm9k1 committed:
fixed Windows backend (for now)

Commit:
GIT_SILENT Upgrade ECM and KF5 version requirements for 5.64.0 release.

dfaure committed:
GIT_SILENT QT_DEPRECATED_WARNINGS_SINCE will be done by ECM

bcooksley committed:
Disable knotification_test, which starts D-Bus if it is not already running, which on Windows causes the test to hang infinitely

Commit:
GIT_SILENT Upgrade KF5 version to 5.65.0.

jbruechert committed:
Document configuration file path on Android

Summary: It's pretty hard right now to find out the search path on android for new developers.

Reviewers: vkrause

Reviewed By: vkrause

Subscribers: kde-frameworks-devel

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D25312

nicolasfella committed:
Make the docs reflect that setIconName should be preferred over setPixmap when possible

Reviewers: #plasma, broulik, apol

Reviewed By: apol

Subscribers: kde-frameworks-devel

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D25396

dfaure committed:
GIT_SILENT enable KF_DISABLE_DEPRECATED_BEFORE_AND_AT=0x053f00 to ensure no use of KF5 deprecated API

romangg committed:
Add missing include for Qt 5.14 build

tm9k1 commented on src/knotification.h:
```diff
@@ -673,6 +673,13 @@ public Q_SLOTS:
+     * @internal
+     * set the id using notification manager
+     */
+    friend class KNotificationManager;
```
To allow `KNotificationManager` class to access `setId()`
The function is not public itself, because it can expose the ID, and that would not be good.

tm9k1 commented on src/knotification.h:
```diff
@@ -673,6 +673,13 @@ public Q_SLOTS:
+     * set the id using notification manager
+     */
+    friend class KNotificationManager;
+    void setId(int id);
```
to set the ID of any notification object to a non -1 ID.

tm9k1 commented on src/notifybysnore.cpp:
```diff
@@ -141,13 +141,6 @@ NotifyBySnore::~NotifyBySnore()
     Q_UNUSED(config);
-    // HACK work around that notification->id() is only populated after returning from here
-    // note that config will be invalid at that point, so we can't pass that along
-    QMetaObject::invokeMethod(this, [this, notification](){ notifyDeferred(notification); }, Qt::QueuedConnection);
```
this hack was not good enough (it caused crashes).

> **To reproduce the crash**: Sync more than 3 notifications at connect.

> **Expected Result** : The notifications show up for first and second notification, but then it crashes. Also, the actions of the notifications do not work.

tm9k1 commented on src/knotificationmanager.cpp:
```diff
@@ -291,6 +291,12 @@ void KNotificationManager::close(int id, bool force)
 
 int KNotificationManager::notify(KNotification *n)
```
I have rewritten the implementation of Notify to fully support the kdeconnect's notification calls.

Commit:
GIT_SILENT Upgrade ECM and KF5 version requirements for 5.65.0 release.

dfaure committed:
GIT_SILENT Bump KF5 deprecation level to 5.64

Commit:
GIT_SILENT Upgrade KF5 version to 5.66.0.

dfaure committed:
GIT_SILENT Bump KF5 deprecation level to 5.65

tosky committed:
Update the obsolete projects.kde.org URL

Use the generic redirect commits.kde.org.

Change discussed on the kde-frameworks-devel list:
https://mail.kde.org/pipermail/kde-frameworks-devel/2019-November/097564.html

tsdgeos committed:
GIT_SILENT Upgrade Qt5 version requirement to 5.12.0.

mirh commented:
So.. Like
If I have to complain about lack of support for notifications in W7, should I open a ticket against knotifications (as in, it is up to that to support every OS) or against snoretoast (as in, that's the specific component which supports Windows, and regardless of whether it uses the actual native api or it just creates toasts on its own)?

Montel committed:
Port endl to \n (QFile flushs data when deleted)

Summary: Port endl to \n (endl is Qt:: namespaced in qt5.15)

Reviewers: dfaure, apol

Reviewed By: apol

Subscribers: kde-frameworks-devel

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D26319

Commit:
GIT_SILENT Upgrade ECM and KF5 version requirements for 5.66.0 release.

davidedmundson committed:
Port KNotification from KWindowSystem

Summary:
KWindowSystem::activateWindow does the same system call as
QWindow::requestActivate for X11 and Windows and both currently do
nothing on wayland.

May as well use Qt implementation

Reviewers: nicolasfella

Reviewed By: nicolasfella

Subscribers: kde-frameworks-devel

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D26513

brute4s99 commented:
I really apologize for this, but we would not be able to add something like that to the upstream codebase, simply because adding support for a dead OS makes sense.

Can I interest you in switching over to Linux with KDE software?
Please take a look at this post from KDE Community: https://twitter.com/ivana_isadora/status/1215409708837888019

STPKITT commented:
I wouldn't call a OS with millions of users dead.

mirh commented:
People make it dead, not microsoft. 
And yes, thank you I already use FOSS where I can.

Commit:
GIT_SILENT Upgrade KF5 version to 5.67.0.

nicolasfella committed:
Deprecate raiseWidget

Summary: It is unused. Furthermore, it relies on QtWidgets API which we try to get rid of in KNotifications. I assume the intention of this method was to raise the window when the notification is clicked, but this should be done by settng a default action and using appropriate QWindow API.

Reviewers: #frameworks, broulik, apol

Reviewed By: apol

Subscribers: kde-frameworks-devel

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D26594

dfaure committed:
GIT_SILENT Bump KF5 deprecation level to 5.66

nicolasfella committed:
Don't use KWindowSystem on Android

Summary:
The usage of KWindowSystem is not relevant on Android. This saves us a dependency there.

Ideally we wouldn't build KPassivePopup at all there, but unfortunately it is public API. One could argue that such a break for non-functional API on Android could be okay.

Depends on D26513

Test Plan: Builds on Android and Linux

Reviewers: #frameworks, vkrause, apol, dfaure

Reviewed By: vkrause, apol, dfaure

Subscribers: dfaure, kde-frameworks-devel

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D26532

kossebau committed:
API dox: teach kapidox & ecm_add_qch about new deprecation macros

GIT_SILENT

kossebau committed:
Add missing declaration of EXCLUDE_DEPRECATED_BEFORE_AND_AT

fix-up for b5ba64a0545d477d212a436398e0aab985603329

GIT_SILENT

kossebau committed:
Exclude also private implementation of raiseWidget if not building

GIT_SILENT

kossebau committed:
API dox: do not use ":" after "@since"

GIT_SILENT
